applications/areg-portal/backend/api/controllers/antibody_controller.py

```diff
@@ -0,0 +1,27 @@
+from typing import List
+
+from api.services import antibody_service
+from openapi.models import AddUpdateAntibody as AddUpdateAntibodyDTO
+from openapi.models import Antibody as AntibodyDTO
+
+
+# TODO: How to handle the http responses given the return types? @afonsobspinto
```

Comment:
@filippomc Do you know the answer to this?
I'm thinking, for example, that there might be a problem with the antibody_controller.create_antibody call, and some exception gets launched. 
I would like to catch that on the controller and return the appropriate HTTP response. 
However, it seems that the main.py is expecting the create_antibody method to return None.
What should we do?

Reply:
@afonsobspinto imo you should raise the "correct" http error e.g. raise a 404 in case you do a get on a non existing object
I would not catch errors so that we can connect sentry to get notified on "unhandled" exceptions

unhandled exceptions are imo bugs and should be resolved via a bugfix / pr

if you catch those exceptions and return a "nice" http response we will not be notified and it will be very difficult to find and fix them

these are my two cents :-)

Reply:
Agree. @zsinnema is Sentry already connected in the cloudharness-django stubs?

@afonsobspinto I guess the confusion comes from the difference between Flask and Fastapi. In Flask controllers exceptions are handled through return values, while in Fastapi exceptions are used https://fastapi.tiangolo.com/tutorial/handling-errors/